src/utils/urlParams/index.ts

```diff
@@ -96,14 +96,33 @@ interface NavigateOptions {
 
 export const glifParams = ['network']
 
+export const isNetworkInPath = (asPath: string) => {
+  // our apps can support both glif.io/<network>/address/<address> AND glif.io/address/<address>/?network=<network>
+  // when we switch networks, we want to make sure we stick with the same routing protocol
+  const supposedNetwork = asPath.split('/')[1]
+  // if the network name was encoded in path we have to rebuild the URL without query params
+  return Object.values(Network).some(
+    net => net === supposedNetwork || net === `${supposedNetwork}net`
+  )
+}
+
 export function navigate(
   router: NextRouter,
   { pageUrl, params, retainParams }: NavigateOptions
 ): void {
   const query = router?.query
   let updatedUrl = pageUrl
 
-  if (query) {
+  if (query && isNetworkInPath(router.asPath)) {
+    updatedUrl += `/${query.network}`
+
+    if (retainParams) {
+      // if the network is already encoded in the url's path, we don't want to add it as a query param
```

Comment:
This could be an issue with other dynamic route parameters, like `address` and `txID` as well, but not sure how to tackle that right now. Not sure if it will be an issue since I don't think any links in the explorer have the need to retain parameters.

Reply:
oh right... hmm will think about this a bit more too